### PR TITLE
Mobile tags

### PR DESCRIPTION
**Pull request by dirkhh**
### Describe the pull request:

- [x] Functional change
- [x] New feature

### Pull request long description:

Calling it a 'new feature' seems a bit excessive - but yeah. We already allowed the user to filter by tags, but we didn't even show tags. That makes no sense.

### Changes made:

- improve vertical 'density' of the DiveDetails - there was just too much empty space
- make tags string available to mobile UI
- add a row to the DiveDetails (but combined with the vertical densification, the overall space needed stayed about the same

### Release note:

done

### Documentation change:

yeah, this will require new pictures and maybe a word or two in the manual

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,53 +1,7 @@
-Desktop: update SAC fields and other statistics when editing cylinders
-Desktop: Reconnect the variations checkbox in planner
-Desktop: add support for dive mode on CSV import and export
-Desktop: fix profile display of planned dives with surface segments
```
Comment (bstoeger) [inline CHANGELOG.md:4]: The top four deleted items are bug fixes, which are not in current release.

File: qt-models/divetripmodel.cpp [hunk 2]
```diff
@@ -222,6 +222,7 @@ QVariant DiveTripModelBase::diveData(const struct dive *d, int column, int role)
 	case MobileListModel::SumWeightRole: return get_weight_string(weight_t { total_weight(d) }, true);
 	case MobileListModel::DiveMasterRole: return d->divemaster ? d->divemaster : QString();
 	case MobileListModel::BuddyRole: return d->buddy ? d->buddy : QString();
+	case MobileListModel::TagsRole: return d->tag_list ? get_taglist_string(d->tag_list) : QString();
```
Comment (bstoeger) [inline qt-models/divetripmodel.cpp]: `get_taglist_string()` should support empty lists. If it doesn't, let's fix `get_taglist_string()`

File: CHANGELOG.md (hunk 1 above)
Comment (dirkhh) [reply CHANGELOG.md:4]: oops. thanks for catching that.

File: qt-models/divetripmodel.cpp (hunk 2 above)
Comment (dirkhh) [reply qt-models/divetripmodel.cpp]: it does. But it's painfully convoluted. It creates a new `struct membuffer`, allocates one byte for the trailing `'\0'`, hands that 'C-string' back to an intermediary function, which then converts it to a `QString` and returns it to us.
But yeah, whatever. It works. Or I could move the `NULL` test one level down into `get_taglist_string()`... I'll take a look.

Comment (dirkhh) [reply qt-models/divetripmodel.cpp]: after reading through this some more and looking at other callers I decided to file this under 'pointless optimizations' and moved on

**Pull request merged**